### PR TITLE
Ospp 2024/feat image dialogue

### PR DESCRIPTION
**Pull request by Fleurxxx**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced image upload capabilities within the chat interface.
	- Added role-based message justification for improved clarity.
	- Conditional rendering of avatars and images based on message sender roles.
	- Implemented image preview and deletion functionality.

- **Styling**
	- Updated styles for chat message images and previews to enhance visual integration.

### CODE REVIEW

File: packages/plugins/robot/src/Main.vue [hunk 1]
```diff
@@ -58,6 +59,9 @@
             >
               <dialog-content :markdownContent="item.content" />
```
Comment (CatsAndMice) [inline packages/plugins/robot/src/Main.vue]: 注意命名规划,中划线

File: packages/plugins/robot/src/Main.vue [hunk 2]
```diff
@@ -283,8 +295,87 @@ export default {
...
+    const imageDeleting = ref(false)
+    const handleDelete = () => {
+      imageDeleting.value = true
+      setTimeout(() => {
+        imageUrl.value = ''
+        imageContent.value = ''
+        imageDeleting.value = false
+        nextTick(() => {
+          if (fileInput.value) {
+            fileInput.value.value = ''
```
Comment (CatsAndMice) [inline packages/plugins/robot/src/Main.vue]: .value.value 可读性不好， 使用unref简化下

File: packages/plugins/robot/src/Main.vue [hunk 3]
```diff
@@ -283,8 +295,87 @@ export default {
...
+    }
+
+    const imageDeleting = ref(false)
+    const handleDelete = () => {
+      imageDeleting.value = true
+      setTimeout(() => {
+        imageUrl.value = ''
+        imageContent.value = ''
+        imageDeleting.value = false
+        nextTick(() => {
```
Comment (CatsAndMice) [inline packages/plugins/robot/src/Main.vue]: setTimeout 为啥还要nextTick？直接setTimeout把 fileInput设为空 可以不？

**Review by CatsAndMice — APPROVED** [review]

File: packages/plugins/robot/src/Main.vue [hunk 4]
```diff
@@ -82,6 +87,11 @@
       <tiny-button @click="endContent">重新发起会话</tiny-button>
       <tiny-button @click="sendContent(inputContent, false)">发送</tiny-button>
     </footer>
+    <input type="file" ref="fileInput" style="display: none" @change="handleFileChange" />
```
Comment (hexqi) [inline packages/plugins/robot/src/Main.vue:90]: 用TinyVue组件库的Input组件

File: packages/plugins/robot/src/Main.vue [hunk 5]
```diff
@@ -237,9 +252,10 @@ export default {
           sessionProcess.displayMessages.push(respDisplayMessage)
           messages.value[messages.value.length - 1].content = originalResponse.choices?.[0]?.message.content
           setContextSession()
-          if (schema?.schema) {
-            createNewPage(schema.schema)
-          }
+          // TODO：返回schema格式的代码
+          // if (schema?.schema) {
```
Comment (hexqi) [inline packages/plugins/robot/src/Main.vue:255]: 这里为什么屏蔽呢

File: packages/plugins/robot/src/Main.vue [hunk 6]
```diff
@@ -283,8 +299,85 @@ export default {
...
+     */
+    const fileInput = ref(null)
+    const openFilePicker = () => {
+      if (unref(fileInput)) {
+        unref(fileInput).click()
+      }
+    }
+
+    let imageUrl = ref('')
+    let imageContent = ref()
```
Comment (hexqi) [inline packages/plugins/robot/src/Main.vue]: 建议用const

File: packages/plugins/robot/src/Main.vue (hunk 4 above)
Comment (CatsAndMice) [reply packages/plugins/robot/src/Main.vue:90]: @Fleurxxx 问题要及时理性掉

File: packages/plugins/robot/src/Main.vue (hunk 5 above)
Comment (Fleurxxx) [reply packages/plugins/robot/src/Main.vue:255]: 因为这里会报错影响这个分支的代码正常运行，在下个分支会进行处理

File: packages/plugins/robot/src/Main.vue (hunk 4 above)
Comment (Fleurxxx) [reply packages/plugins/robot/src/Main.vue:90]: TinyVue组件库的Input组件不直接支持文件上传，所以用原生input作为文件输入元素来选择图片

**Review by hexqi — APPROVED** [review]

**Review by hexqi — CHANGES_REQUESTED** [review]
@Fleurxxx  部分commit没有签名，可以使用这条命令(末尾commitId替换成起始commit id)：
`git rebase --exec 'git commit --amend --no-edit -n -S' -i commitId`
会对到-i之前的 commit 为止 (包括 - i) 所有的 commit 都会执行一次git commit --amend --no-edit -n -S来签名

**Pull request closed without merging**